care/facility/api/viewsets/facility.py

```diff
@@ -71,8 +71,15 @@ class FacilityViewSet(
 ):
     """Viewset for facility CRUD operations."""
 
-    queryset = Facility.objects.all().select_related(
-        "ward", "local_body", "district", "state"
+    queryset = (
+        Facility.objects.all()
+        .select_related(
+            "ward",
+            "local_body",
+            "district",
+            "state",
+        )
+        .prefetch_related("patientregistration_set", "bed_set")
```

Comment:
instead of prefetch try annotating the count using subquery, and in the serializer check if the annotated attribute is present else fallback to filtering